### PR TITLE
build: upgrade webpack, jest, babel and other pipeline tools

### PR DESCRIPTION
**Pull request by eszthoff**
Story: [ONEUI-177](https://jira.textkernel.nl/browse/ONEUI-177)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,80 +68,77 @@
...
+        "ts-loader": "^6.2.2",
         "typescript": "^3.8.3",
-        "webpack": "4.28.1",
-        "webpack-bundle-analyzer": "3.3.2",
-        "webpack-cli": "3.2.1",
-        "webpack-dev-server": "3.1.14"
+        "webpack": "^4.42.1",
+        "webpack-bundle-analyzer": "^3.6.1",
+        "webpack-cli": "^3.3.11",
+        "webpack-dev-server": "^3.10.3"
```
Comment (mukiienko) [inline package.json:142]: What do you think about updating other libraries? 
Screenshot 2020-04-01 at 11 00 40

File: package.json [hunk 2]
```diff
@@ -69,80 +68,77 @@
...
+        "ts-jest": "^24.3.0",
+        "ts-loader": "^6.2.2",
         "typescript": "^3.8.3",
-        "webpack": "4.28.1",
-        "webpack-bundle-analyzer": "3.3.2",
-        "webpack-cli": "3.2.1",
-        "webpack-dev-server": "3.1.14"
+        "webpack": "^4.42.1",
+        "webpack-bundle-analyzer": "^3.6.1",
+        "webpack-cli": "^3.3.11",
```
Comment (mukiienko) [inline package.json:141]: I think better to keep a fixed version for all dependencies or at least patch releases `~`.
I remember we have problems with that when some library version was increased automatically (probably after npm install).

Comment (carlobernardini) [reply package.json:141]: Relevant read about (not) pinning dependencies: https://medium.com/the-guild/how-should-you-pin-your-npm-dependencies-and-why-2b8d545c7312

File: package.json [hunk 3]
```diff
@@ -34,7 +34,7 @@
     },
     "husky": {
         "hooks": {
-            "pre-commit": "npm-run-all --parallel tsc test lint:*"
+            "pre-commit": "npm-run-all --parallel tsc test lint:ci lint:styles"
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply package.json:37]: the new husky didn't like it.

File: package.json (hunk 1 above)
Comment (eszthoff) [reply package.json:142]: next subtask

File: package.json (hunk 2 above)
Comment (eszthoff) [reply package.json:141]: shall we bring it to the meeting tomorrow?

**Pull request merged**